services/server/src/server/services/storageServices/SourcifyDatabaseService.ts

```diff
@@ -144,13 +144,25 @@ export class SourcifyDatabaseService
         creationMatch: match.creationMatch,
       });
     } else if (type === "update") {
-      if (verifiedContractId) {
-        await Database.updateSourcifyMatch(this.databasePool, {
+      if (!verifiedContractId) {
+        throw new Error(
```

Comment:
@kuzdogan do we need to add a logger.error() before throwing?

Reply:
Anyting we want to be alerted about needs to be `logger.error` logged. If that's the case yes.